feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/microprofile/opentracing-smallrye/main/module.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2018 Red Hat, Inc. and/or its affiliates
+  ~ and other contributors as indicated by the @author tags.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<module xmlns="urn:jboss:module:1.8" name="org.wildfly.extension.microprofile.opentracing-smallrye">
+    <resources>
+        <artifact name="${org.wildfly:wildfly-microprofile-opentracing-smallrye}"/>
+
+        <artifact name="${io.smallrye:smallrye-opentracing}"/>
+
+        <artifact name="${io.opentracing:opentracing-api}"/>
+        <artifact name="${io.opentracing:opentracing-noop}"/>
+        <artifact name="${io.opentracing:opentracing-util}"/>
+        <artifact name="${io.opentracing.contrib:opentracing-concurrent}"/>
+        <artifact name="${io.opentracing.contrib:opentracing-jaxrs2}"/>
+        <artifact name="${io.opentracing.contrib:opentracing-web-servlet-filter}"/>
+
+        <artifact name="${io.opentracing.contrib:opentracing-tracerresolver}"/>
+        <artifact name="${io.jaegertracing:jaeger-core}"/>
+        <artifact name="${io.jaegertracing:jaeger-thrift}"/>
+
+        <artifact name="${com.google.code.gson:gson}"/>
+    </resources>
```

Comment:
It's unusual to have this many artifacts in the same module. Generally these should be modules on their own that `org.wildfly.extension.microprofile.opentracing-smallrye` depends on.

Reply:
+1 . Now that we have everything working and TCK passing, I was able to move things into their own modules and got it almost working (works on some tests already). The current problem is that I have two classes under the same package, but different modules, due to https://github.com/opentracing-contrib/java-tracerresolver/issues/31 . I have this worked saved in a different branch and will resume working on it if we can get a fix/release of ot-tracerresolver in time.